### ISSUE TITLE
Fastlane match getting stuck up when cloning from bitbucket

### ISSUE THREAD

**Issue by santoshshetty2612** [fastlane/fastlane#6833]
Hi,

I never faced this issue on my build server which is on a very good network. But since morning today am constantly facing this issue.

Sometimes the cloning and fetching of profiles works fine. But most of the time it gets stuck and finally fails :

INFO [2016-11-02 18:08:26.08]: Cloning remote git repo...
INFO [2016-11-02 18:08:26.08]: [4m[36m$ GIT_TERMINAL_PROMPT=0 git clone 'git@bitbucket.org:myproject/myprofiles.git' '/var/folders/fc/lp0tsgx51_ldd7m_v7_zqnc800007q/T/d20161102-89649-1xyj7os'[0m[0m
INFO [2016-11-02 18:08:26.09]: ▸ [35mCloning into '/var/folders/fc/lp0tsgx51_ldd7m_v7_zqnc800007q/T/d20161102-89649-1xyj7os'...[0m
INFO [2016-11-02 18:08:29.29]: ▸ [35mremote: Counting objects: 12474, done.[K[0m
INFO [2016-11-02 18:08:30.94]: ▸ [35mremote: Compressing objects: 100% (11539/11539), done.[K[0m

**It fails with the below error:**
Connection to bitbucket.org closed by remote host.
fatal: The remote end hung up unexpectedly
fatal: early EOF
fatal: index-pack failed

**Comment by TKBurner** [comment]
Hi @santoshshetty2612 Can you run `fastlane env` and share the output? It looks like this could be an issue with your permissions on bitbucket but I want to investigate. Thanks!

**Comment by santoshshetty2612** [comment]
Hi @TKBurner, please find the output of fastlane env.

<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                 | Value                                                   |
| ------------------- | ------------------------------------------------------- |
| OS                  | 10.12                                                   |
| Ruby                | 2.3.0                                                   |
| Bundler?            | false                                                   |
| Xcode Path          | /Applications/Xcode.app/Contents/Developer/             |
| Xcode Version       | 8.0                                                     |
| Git                 | git version 2.8.4 (Apple Git-73)                        |
| Installation Source | /Users/Shared/Jenkins/.rvm/gems/ruby-2.3.0/bin/fastlane |
| Host                | Mac OS X 10.12 (16A323)                                 |
| Ruby Lib Dir        | /Users/Shared/Jenkins/.rvm/rubies/ruby-2.3.0/lib        |
| OpenSSL Version     | OpenSSL 1.0.2j  26 Sep 2016                             |


### fastlane files:

**No Fastfile found**


**No Appfile found**


### fastlane gems

| Gem                 | Version | Update-Status       |
| ------------------- | ------- | ------------------- |
| credentials_manager | 0.16.2  | ✅ Up-To-Date        |
| fastlane_core       | 0.53.0  | ✅ Up-To-Date        |
| spaceship           | 0.36.2  | ✅ Up-To-Date        |
| snapshot            | 1.16.2  | 🚫 Update availaible |
| frameit             | 3.0.0   | ✅ Up-To-Date        |
| pem                 | 1.3.2   | 🚫 Update availaible |
| cert                | 1.4.3   | ✅ Up-To-Date        |
| sigh                | 1.11.2  | ✅ Up-To-Date        |
| produce             | 1.2.1   | ✅ Up-To-Date        |
| gym                 | 1.11.3  | ✅ Up-To-Date        |
| pilot               | 1.11.1  | 🚫 Update availaible |
| scan                | 0.13.1  | ✅ Up-To-Date        |
| supply              | 0.7.1   | ✅ Up-To-Date        |
| match               | 0.10.0  | ✅ Up-To-Date        |
| screengrab          | 0.5.3   | 🚫 Update availaible |
| fastlane            | 1.106.2 | ✅ Up-To-Date        |


### Loaded fastlane plugins:

**No plugins Loaded***


<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.0.0        |
| executable-hooks          | 1.3.2        |
| bundler-unload            | 1.0.2        |
| rubygems-bundler          | 1.4.4        |
| bundler                   | 1.12.5       |
| io-console                | 0.4.5        |
| highline                  | 1.7.8        |
| commander                 | 4.4.0        |
| terminal-table            | 1.4.5        |
| json                      | 1.8.3        |
| multipart-post            | 2.0.0        |
| faraday                   | 0.9.2        |
| faraday_middleware        | 0.10.0       |
| dotenv                    | 2.1.1        |
| net-ssh                   | 3.2.0        |
| net-sftp                  | 2.1.2        |
| plist                     | 3.1.0        |
| rubyzip                   | 1.1.7        |
| security                  | 0.1.3        |
| krausefx-shenzhen         | 0.14.10      |
| slack-notifier            | 1.5.1        |
| i18n                      | 0.7.0        |
| thread_safe               | 0.3.5        |
| tzinfo                    | 1.2.2        |
| activesupport             | 4.2.7.1      |
| colored                   | 1.2          |
| claide                    | 1.0.0        |
| xcodeproj                 | 1.2.0        |
| rouge                     | 1.11.1       |
| xcpretty                  | 0.2.4        |
| terminal-notifier         | 1.6.3        |
| addressable               | 2.3.8        |
| xcode-install             | 2.0.2        |
| word_wrap                 | 1.0.0        |
| multi_json                | 1.12.1       |
| babosa                    | 1.0.2        |
| excon                     | 0.54.0       |
| gh_inspector              | 1.0.2        |
| credentials_manager       | 0.16.2       |
| fastlane_core             | 0.53.0       |
| multi_xml                 | 0.5.5        |
| unf_ext                   | 0.0.7.2      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20160615 |
| http-cookie               | 1.0.2        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 1.6.8        |
| mini_magick               | 4.5.1        |
| xcpretty-travis-formatter | 0.0.4        |
| uber                      | 0.0.15       |
| representable             | 2.3.0        |
| retriable                 | 2.1.0        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| hurley                    | 0.2          |
| little-plugger            | 1.1.4        |
| logging                   | 2.1.0        |
| jwt                       | 1.5.4        |
| memoist                   | 0.15.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.1        |
| thor                      | 0.19.1       |
| httpclient                | 2.8.2.2      |
| google-api-client         | 0.9.12       |
</details>


*generated on:* **2016-11-03**
</details>

**Comment by santoshshetty2612** [comment]
I am cloning multiple profiles. It works fine for some profiles but get stuck up randomly.

**Comment by santoshshetty2612** [comment]
private_lane :syncprofiles_for_app_identifier do |options|
	match(type: "development", readonly: !options[:admin], app_identifier: options[:app_identifier], force_for_new_devices: true)
	sleep 4
	match(type: "adhoc", readonly: !options[:admin], app_identifier: options[:app_identifier], force_for_new_devices: true)
	sleep 4
	match(type: "appstore", readonly: !options[:admin], app_identifier: options[:app_identifier], force_for_new_devices: true)
	sleep 4
end

Further updates to the problem am facing. I am using the above private lane to generate the profiles. I have around 4 app Identifiers. For every app Id, I call the above private lane. I tried solving the problem by adding a  sleep of 4 seconds between calling match assuming bitbucket might be causing problems because of too fast communication. But still the problem exists.

match will work absolutely fine to create few profiles. But randomly gets hung at:

INFO [2016-11-03 19:04:49.13]: ▸ [35mremote: Counting objects: 12999, done.[K[0m
INFO [2016-11-03 19:04:51.69]: ▸ [35mremote: Compressing objects: 100% (12063/12063), done.[K[0m

**Comment by santoshshetty2612** [comment]
I found the reason for the issue. 
The issue is with Bitbucket. Even through command line when we try to clone the repo it stucks sometime though this frequency is less.
I tried both ssh and https, but the result was same.

I have shifted to gitlab to store the match certificates and profiles. gitlab is not as fast as bitbucket when cloning and pushing to the repo but is behaving quite consistent till now.

Hence closing this issue.
